### PR TITLE
Refactored controller advice so API callers receive more consistent, more manageable errors

### LINKED ISSUE #10: A Good Way(TM) of passing error codes/causes back to front ends
The current controller advice pattern makes it difficult for front end developers to parse rejected API responses.

Ideally, we'd return a known response structure on bad requests which includes an error code that can be internationalised on the front end.

### PR DESCRIPTION
**Pull request by sluehr**

### CODE REVIEW

**Review by hriteshg — APPROVED** [review]

**Comment by sluehr** [comment]
Addressed #10

File: api/src/test/java/engineering/everest/lhotse/api/rest/ExceptionHandlingControllerAdviceTest.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+                .timeStamp(Instant.now(clock))
+                .build();
+
+        assertEquals(new ResponseEntity<>(expectedResponse, new HttpHeaders(), BAD_REQUEST), controllerAdvice.handleRemoteCommandExecutionException(exception));
+    }
+
+    @Test
+    void willMapMethodArgumentNotValidExceptions() throws Exception {
+        var bindingResult = mock(BeanPropertyBindingResult.class);
+        var errors = List.of(
```
Comment (hriteshg) [inline api/src/test/java/engineering/everest/lhotse/api/rest/ExceptionHandlingControllerAdviceTest.java:174]: Static import.

**Review by hriteshg — APPROVED** [review]

File: api/src/test/java/engineering/everest/lhotse/api/rest/ExceptionHandlingControllerAdviceTest.java (hunk 1 above)
Comment (sluehr) [reply api/src/test/java/engineering/everest/lhotse/api/rest/ExceptionHandlingControllerAdviceTest.java:174]: Not in this case; List.of() is readable, of() is semantically ... lost.

**Review by ywangd — APPROVED** [review]
I feel BadRequest has been used as a catch-all error. One could argue that exceptions like RemoteCommandExecutionException should be `500`. But I think this is not the point of this PR. So I think it is good to merge.

**Comment by sluehr** [comment]
I'll merge this but let's have a conversation on what to do with 500 errors and whether we want to expose them (albeit with reduced reporting of application internals).

**Review by ywangd — APPROVED** [review]
Looks good to me.

**Pull request merged**